### PR TITLE
Implement Sleep Duration & Date Tracking with ISO Date Standardization and  Search Feature

### LINKED ISSUE #126: Standardize sleep to use ISO format
Currently sleep functionality classes and methods use dd-MM-YYYY HH:MM to parse and print datetime. For consistency reasons, standardization to YYYY-MM-dd HH:MM (ISO Date format) is required to be implemented.

### LINKED ISSUE #137: Sleep Duration and Sleep Date Tracking
Implement a feature to track and record sleep duration along with the associated date. Account for sleep sessions that begin after midnight but before 6am, attributing them to the previous day's date.

### PR DESCRIPTION
**Pull request by DaDevChia**
1. Closes #137 
2. Closes #126 

- Standardized sleep datetime format to use ISO Date format of YYYY-MM-d.
- Implemented Sleep Duration and Date tracking functionality such that users can know how long and on which day they have slept.
- Added Find functionality for sleep to search sleep record for the day itself.
- Added helper methods to prepare for implementation of Sleep Goal tracking.
- Updated Developer Guide on sleep and sleeplist class diagram.

### CODE REVIEW

File: src/main/java/athleticli/ui/Parser.java [hunk 1]
```diff
@@ -710,32 +706,26 @@ public static void checkMissingActivityGoalArguments(int sportIndex, int targetI
      */
     public static AddSleepCommand parseSleepAdd(String commandArgs) throws AthletiException {
 
-        int startMarkerPos = commandArgs.indexOf(Parameter.START_TIME_SEPARATOR);
-        int endMarkerPos = commandArgs.indexOf(Parameter.END_TIME_SEPARATOR);
+        final int startTimeIndex = commandArgs.indexOf(Parameter.START_TIME_SEPARATOR);
+        final int endTimeIndex = commandArgs.indexOf(Parameter.END_TIME_SEPARATOR);
```
Comment (skylee03) [inline src/main/java/athleticli/ui/Parser.java:710]: Great use of `final`!

File: src/main/java/athleticli/data/sleep/Sleep.java [hunk 2]
```diff
@@ -1,35 +1,118 @@
...
+    }
+
+    /**
+     * Calculate the sleep date based on start time.
+     * Factor in the possibility of sleeping past midnight.
+     * We are assuming that the user sleeps before 6am even if the user sleeps past midnight.
+     * @return sleep date.
+     */
+    private LocalDate calculateSleepDate() {
+        if (startDateTime.getHour() < 6) {
```
Comment (skylee03) [inline src/main/java/athleticli/data/sleep/Sleep.java:75]: It takes good care of the sleeping habits of actual users!

**Review by skylee03 — APPROVED** [review]
LGTM!

File: src/main/java/athleticli/data/sleep/Sleep.java (hunk 2 above)
Comment (DaDevChia) [reply src/main/java/athleticli/data/sleep/Sleep.java:75]: Thanks!

**Pull request merged**